modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/service/impl/SXPElementLocalServiceImpl.java

```diff
@@ -133,6 +152,49 @@ public List<SXPElement> getSXPElements(long companyId, boolean readOnly) {
 		return sxpElementPersistence.findByC_R(companyId, readOnly);
 	}
 
+	@Override
+	public void importSXPElements(Company company) throws PortalException {
+		Set<String> externalReferenceCodes = new HashSet<>();
+
+		for (SXPElement sxpElement :
+				getSXPElements(company.getCompanyId(), true)) {
+
+			externalReferenceCodes.add(sxpElement.getExternalReferenceCode());
+		}
+
+		for (com.liferay.search.experiences.rest.dto.v1_0.SXPElement
+				sxpElement : _getSXPElements()) {
+
+			if ((!FeatureFlagManagerUtil.isEnabled("LPS-122920") &&
```

Comment:
@MrJohn1911 Please check if we can move this feature flag check to the beginning of this method.